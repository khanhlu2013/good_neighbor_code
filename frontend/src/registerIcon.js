import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faThumbsUp, //approve
  faThumbsDown, //deny
  faTrashAlt, //remove
  faUndoAlt,
  faPlay, //play youtube video
  faPause, //pause youtube video
  faClock, //youtube video duration
  faBriefcase, //my post
  faUserCog, //profile image replacer
  faGlobe, //in posts
  faUserFriends, //friends or my networks
  faHandHoldingHeart, //borrow (work both for in and out post)
  faHistory,
  faCheck, //approve : when request is being approve, could also use for isActivePost.
  faQuestion, //request: making a request from inPost
  faRetweet,
  faSignOutAlt, //connection - my request
  faSignInAlt, //connection - friend request
  faSearch,
  faUserSlash //deny user list
} from "@fortawesome/free-solid-svg-icons";

library.add(
  faThumbsUp,
  faThumbsDown,
  faTrashAlt,
  faUndoAlt,
  faPlay,
  faPause,
  faClock,
  faBriefcase,
  faUserCog,
  faGlobe,
  faUserFriends,
  faHandHoldingHeart,
  faHistory,
  faCheck,
  faQuestion,
  faRetweet,
  faSignOutAlt,
  faSignInAlt,
  faSearch,
  faUserSlash
);
