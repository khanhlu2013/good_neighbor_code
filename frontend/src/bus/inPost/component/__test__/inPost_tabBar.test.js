import React from "react";
import { shallow, mount } from "enzyme";

import InPostTabBar from "../inPost_tabBar";
import InPostTabEnum from "../inPost_tabEnum";
import "../../../../registerIcon";

describe("inPost_tabBar", () => {
  it("to trigger onTabChange callback handler correctly", () => {
    const onTabChange = jest.fn();
    const wrap = mount(
      <InPostTabBar
        selectTab={InPostTabEnum.ALL}
        onTabChange={onTabChange}
        allCount={0}
        requestCount={1}
        approveCount={2}
        borrowCount={3}
        historyCount={4}
      />
    );

    // expect(wrap).toMatchSnapshot();
    wrap.find("div#tabSelector-inPost-all-react").simulate("click");
    wrap.find("div#tabSelector-inPost-request-react").simulate("click");
    wrap.find("div#tabSelector-inPost-approve-react").simulate("click");
    wrap.find("div#tabSelector-inPost-borrow-react").simulate("click");
    wrap.find("div#tabSelector-inPost-history-react").simulate("click");

    expect(onTabChange.mock.calls.length).toBe(5);

    expect(onTabChange.mock.calls[0][0]).toBe(InPostTabEnum.ALL);
    expect(onTabChange.mock.calls[1][0]).toBe(InPostTabEnum.REQUEST);
    expect(onTabChange.mock.calls[2][0]).toBe(InPostTabEnum.APPROVE);
    expect(onTabChange.mock.calls[3][0]).toBe(InPostTabEnum.BORROW);
    expect(onTabChange.mock.calls[4][0]).toBe(InPostTabEnum.HISTORY);
  });

  it("to match snapshot", () => {
    const onTabChange = jest.fn();
    const wrap = shallow(
      <InPostTabBar
        selectTab={InPostTabEnum.ALL}
        onTabChange={onTabChange}
        allCount={0}
        requestCount={1}
        approveCount={2}
        borrowCount={3}
        historyCount={4}
      />
    );
    expect(wrap).toMatchSnapshot();
  });
});
