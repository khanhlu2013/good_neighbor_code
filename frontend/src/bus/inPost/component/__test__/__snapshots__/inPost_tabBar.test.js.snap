// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inPost_tabBar to match snapshot 1`] = `
<styled.div>
  <AppBodyTabItem
    caption="all friend posts"
    iconName="globe"
    id="tabSelector-inPost-all-react"
    isResponsive={false}
    isSelect={true}
    notificationItem={
      <AlertItem
        count={0}
        isImportant={false}
      />
    }
    onSelect={[Function]}
  />
  <AppBodyTabItem
    caption="request"
    iconName="question"
    id="tabSelector-inPost-request-react"
    isResponsive={true}
    isSelect={false}
    notificationItem={
      <AlertItem
        count={1}
        isImportant={false}
      />
    }
    onSelect={[Function]}
  />
  <AppBodyTabItem
    caption="approve"
    iconName="check"
    id="tabSelector-inPost-approve-react"
    isResponsive={true}
    isSelect={false}
    notificationItem={
      <AlertItem
        count={2}
        isImportant={true}
      />
    }
    onSelect={[Function]}
  />
  <AppBodyTabItem
    caption="borrow"
    iconName="hand-holding-heart"
    id="tabSelector-inPost-borrow-react"
    isResponsive={true}
    isSelect={false}
    notificationItem={
      <AlertItem
        count={3}
        isImportant={false}
      />
    }
    onSelect={[Function]}
  />
  <AppBodyTabItem
    caption="history"
    iconName="history"
    id="tabSelector-inPost-history-react"
    isResponsive={true}
    isSelect={false}
    notificationItem={
      <AlertItem
        count={4}
        isImportant={false}
      />
    }
    onSelect={[Function]}
  />
</styled.div>
`;
